File: Examples/test-suite/php/director_profile_runme.php
<?php

require "tests.php";

// No new functions
check::functions(array());
// New classes
check::classes(array('A','B'));
// No new vars
check::globals(array());

class MyB extends B {
  function vfi($a) {
    return $a+3;
  }
}

$a = new A();
$myb = new MyB();
$b = B::get_self($myb);

$i = 50000;
$a = 1;

while ($i) {
  $a = $b->fi($a); #1
  $a = $b->fi($a); #2
  $a = $b->fi($a); #3
  $a = $b->fi($a); #4
  $a = $b->fi($a); #5
  $a = $b->fi($a); #6
  $a = $b->fi($a); #7
  $a = $b->fi($a); #8
  $a = $b->fi($a); #9
  $a = $b->fi($a); #10
  $a = $b->fi($a); #1
  $a = $b->fi($a); #2
  $a = $b->fi($a); #3
  $a = $b->fi($a); #4
  $a = $b->fi($a); #5
  $a = $b->fi($a); #6
  $a = $b->fi($a); #7
  $a = $b->fi($a); #8
  $a = $b->fi($a); #9
  $a = $b->fi($a); #20
  $i -= 1;
}

print $a . "\n";

check::done();

File: Examples/test-suite/lua/director_profile_runme.lua
require("import") -- the import fn
import("director_profile") -- import lib

local myb = director_profile.B()
function vfi(self, a)
    return a + 3
end
swig_override(myb, 'vfi', vfi)
local b = director_profile.B.get_self(myb)

local fi = b.fi
local a = 1

for i = 1,50000 do
    a = fi(b, a) -- 1
    a = fi(b, a) -- 2
    a = fi(b, a) -- 3
    a = fi(b, a) -- 4
    a = fi(b, a) -- 5
    a = fi(b, a) -- 6
    a = fi(b, a) -- 7
    a = fi(b, a) -- 8
    a = fi(b, a) -- 9
    a = fi(b, a) -- 10
    a = fi(b, a) -- 1
    a = fi(b, a) -- 2
    a = fi(b, a) -- 3
    a = fi(b, a) -- 4
    a = fi(b, a) -- 5
    a = fi(b, a) -- 6
    a = fi(b, a) -- 7
    a = fi(b, a) -- 8
    a = fi(b, a) -- 9
    a = fi(b, a) -- 20
end

if false then
    print("a:", a)
end
